api/all.yaml

```diff
@@ -722,6 +748,268 @@ paths:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateModelVersion'
+      responses:
+        '200':
+          description: The new model version was successfully created.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ModelVersionInfo'
+  /models/{full_name}/versions:
+    parameters:
+      - name: full_name
+        in: path
+        description: Full name of the model.
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - ModelVersions
+      operationId: listModelVersions
+      summary: List model versions
+      description: |
+        Gets the list of all available model versions under the model.
+        There is no guarantee of a specific ordering of the elements in the array.
+      responses:
+        '200':
+          description: The model versions list was successfully retrieved.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListModelVersionsResponse'
+  /models/{full_name}/versions/{version}:
+    parameters:
+      - name: full_name
+        in: path
+        description: Full name of the model.
+        required: true
+        schema:
+          type: string
+      - name: version
+        in: path
+        description: Version number of the model version.
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - ModelVersions
+      operationId: getModelVersion
+      summary: Get a model version
+      description: |
+        Gets a specific model version for a specific model.
+      responses:
+        '200':
+          description: The model version was successfully retrieved.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ModelVersionInfo'
+    patch:
+      tags:
+        - ModelVersions
+      operationId: updateModelVersion
+      summary: Update a model version
+      description: |
+        Updates the model version that matches the supplied name.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UpdateModelVersion'
+      responses:
+        '200':
+          description: The model version was successfully updated.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ModelVersionInfo'
+    delete:
+      tags:
+        - ModelVersions
+      operationId: deleteModelVersion
+      summary: Delete a model version
+      description: |
+        Deletes a model version from the specified model.
+      responses:
+        '200':
+          description: The model was successfully deleted.
+          content:
+            application/json:
+              schema: {}
+  /models/{full_name}/versions/{version}/finalize:
```

Comment:
We need the finalize API in UC OSS as well? Is this mainly to maintain parity since I imagine most folks won't need this API in their custom implementations?

Reply:
Yes.  How else will you tell when the artifacts have been fully uploaded to the storage location?

Reply:
Yup, makes sense. Not sure how public-facing these docs will be, but I can imagine some users being confused at that, so adding that bit about `artifacts have been fully uploaded to the storage location` for this API will probably help them implement it accordingly

Reply:
It's public undocumented currently.  Its on my list of things to track down how to handle because the credential endpoint is also the same way.  We will probably need to document it in DB UC if we are adding it into UC OSS.  I'll sort that out with @haogang and @tdas